roaring/roaring.go

```diff
@@ -705,93 +674,98 @@ func (b *Bitmap) unionInPlace(others ...*Bitmap) {
 				continue
 			}
 
-			var (
-				iKey, iContainer = iIter.iter.Value()
-				// Summary statistics about all the containers in the other bitmaps
-				// that share the same key so we can make smarter union strategy
-				// decisions later. Note that we slice to [i:] not [i+1:] because we
-				// want to include the current containers information in the stats.
-				summaryStats = bitmapIters[i:].calculateSummaryStats(iKey)
-			)
-
-			if summaryStats.isOnlyContainerWithKey {
-				// TODO(rartoul): We can avoid these clones if we can determine
-				// if the container is coming from an immutable bitmap and we
-				// know that we can mark the target bitmap as immutable as well.
-				target.Containers.Put(iKey, iContainer.Clone())
-				bitmapIters[i].handled = true
-				continue
-			}
+			iKey, iContainer := iIter.iter.Value()
+			expectedN := int64(0)
 
-			// There was more than one container across the bitmaps with key iKey
-			// so we need to calculate a union.
+			// determine whether we have a target to union into.
+			tContainer := target.Containers.Get(iKey)
+			// if the target's full, short-circuit out.
+			if tContainer != nil {
+				if tContainer.n == maxContainerVal+1 {
+					bitmapIters.markItersWithCurrentKeyAsHandled(i, iKey)
```

Comment:
So kind of a jerk move to ask this since I originally named this function, but looking at it now seems like the name should be `markItersWithKeyAsHandled` since you have to pass in the key, its not relying on any iterator state right?